elements/fsm-builder/fsm-builder.js

```diff
@@ -0,0 +1,646 @@
+
+//var canvas;
+
+//var caretTimer;
+//var caretVisible = true;
+
+//var nodes = [];
+//var links = [];
+
+//var stateNamesToHighlight = null
+//var transitionsToHighlight = null
+
+const snapToPadding = 6; // pixels
```

Comment:
My main comment at this point: let's keep as much as possible out of the global scope. That goes for these constants, as well as all the classes in the JS files in `fsm-links.js`. You can do this by moving everything to one file, wrapping it in an IIFE, and then manually exporting only the bits that should be externally-accessible to the `window` object.

Reply:
@eliotwrobson and I had a good chat the other day about different ways to address this. I should have commented here to note, sorry.

Reply:
Just did this refactor (only took a few minutes). I _think_ I did it right 😅